aws_lambda_builders/workflows/nodejs_npm/npm.py

```diff
@@ -0,0 +1,53 @@
+"""
+Wrapper around calling npm through a subprocess.
+"""
+
+import logging
+
+from .utils import OSUtils
+
+LOG = logging.getLogger(__name__)
+
+
+class NpmError(Exception):
+    pass
+
+
+class NpmExecutionError(NpmError):
+    def __init__(self, err):
+        super(NpmExecutionError, self).__init__(
+            'NPM failed: %s' % err)
+
+
+class SubprocessNpm(object):
+
+    def __init__(self, osutils=None, npm_exe=None):
+        if osutils is None:
+            osutils = OSUtils()
+        self.osutils = osutils
+
+        if npm_exe is None:
+            npm_exe = 'npm'
+
+        self.npm_exe = npm_exe
+
+    def main(self, args, cwd=None):
```

Comment:
you can simplify this further by using this neat Python trick:

`run(*args, cwd=None)` - then your caller can call this method using `run("install", "--production", cwd='blah')`, but you will end up getting a list value within the `args` variable that you can call Popen with.

Not that you should make this change, but felt like a neat Py trick you might like :)

Reply:
thanks for the tip. I'd prefer to leave it like this because it's a single list to pass on, it will make it easier to switch to user-provided values when we introduce the env var to override default arguments. (as in that case the first element would have to stay the same, but the users will be able to override other values)